### PR TITLE
Feature/state management

### PR DESCRIPTION
**Pull request by AndriiSameliuk**
Implements Redux state management approach to manage events.
image

### CODE REVIEW

**Comment by AndriiSameliuk** [comment]
@dkalinin, PTAL.

File: app/Controller/FileListController/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  constructor(root, store) {
+    super(root, store);
+    this._fetchFiles();
+  }
+
+  /**
+   * Creates page which contains files.
+   * @private
+   */
+  _fetchFiles() {
```
Comment (dkalinin) [inline app/Controller/FileListController/index.js]: _createPage

File: app/View/Component/ConfiguringComponent/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  /**
+   * Creates object.
+   * @public
+   * @param {HTMLElement} root - root element of html page.
+   * @param {Store} store
+   */
+  constructor(root, store) {
+    super(root);
+    this.store = store;
+    this.store.events.subscribe('stateChange', () => this.initRender());
```
Comment (dkalinin) [inline app/View/Component/ConfiguringComponent/index.js]: Let's call `render`

File: app/View/Component/FileListForm/index.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    super(root, store);
+    this.store.dispatch('fetchFiles');
+  }
+
+  /**
+   * Creates span which show to user that loading continue.
+   * @public
+   */
+  initRender() {
+    const span = Component.createElement(this.root, 'upload-span', 'span');
```
Comment (dkalinin) [inline app/View/Component/FileListForm/index.js]: Span should be created only if we are going to show text (if isFilesLoading === true).

File: app/View/Component/FileListPage/index.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  /**
+   * Create list of files.
+   * @public
+   */
+  initRender() {
+    const div = document.createElement('div');
+    this.root.appendChild(div);
+    new FileListForm(div, this.store);
```
Comment (dkalinin) [inline app/View/Component/FileListPage/index.js]: FileList

File: app/View/Application/index.js [hunk 5]
```diff
@@ -2,6 +2,12 @@ import {Router} from '../../Router';
 import {RegistrationPageController} from '../../Controller/RegistrationPageController';
 import {AuthorizationPageController} from '../../Controller/AuthorizationPageController';
 import {Page404Controller} from '../../Controller/Page404Controller';
+import {FileListController} from '../../Controller/FileListController';
+import {Store} from '../../Store';
+import mutations from '../../Store/Mutations';
+import actions from '../../Store/Actions';
+import states from '../../Store/State';
```
Comment (dkalinin) [inline app/View/Application/index.js]: Please rename to state

**Comment by AndriiSameliuk** [comment]
@dkalinin, fixed, PTAL.

File: app/Store/index.js [hunk 6]
```diff
@@ -30,7 +30,7 @@ export class Store {
    * @param {String} name
    * @param {Object} value
    */
-  dispatch(name, value) {
+  dispatch(name, value = {}) {
```
Comment (dkalinin) [inline app/Store/index.js]: It should not have a default value. It can be like:
```js
dispatch(actionName, ...params)
```

Comment (dkalinin) [reply app/Store/index.js]: Check [here](https://stackoverflow.com/questions/4729516/correct-way-to-document-open-ended-argument-functions-in-jsdoc) a way to write a doc for it.

File: app/View/Component/ConfiguringComponent/index.js [hunk 7]
```diff
@@ -8,11 +9,27 @@ export class ConfiguringComponent extends Component {
...
+  get state() {
+    return this.store.state;
+  }
+
+  /**
+   * Dispatches events.
+   * @param {String} name
+   * @param {Object} value
+   */
+  dispatch(name, value = {}) {
```
Comment (dkalinin) [inline app/View/Component/ConfiguringComponent/index.js]: The same comment as for Store.dispatch().

**Pull request merged**